java/src/jmri/jmrit/roster/Roster.java

```diff
@@ -321,6 +321,23 @@ public RosterEntry getEntryForId(String id) {
         return null;
     }
 
+     /**
+     * Return a list of RosterEntry which have a 
+     * particular DCC address.
+     *
+     * @param a 
+     * @return an ArrayList of matching entries
+     */
+    public List<RosterEntry> getEntriesByDccAddress(String a) {
+        List<RosterEntry> list = new ArrayList<>(); 
```

Comment:
On Fri, Mar 18, 2016 at 8:59 AM, Steve Rawlinson <notifications@github.com>
wrote:

> I'm not sure I agree. The end result is the same but matchingList() does
> an awful lot of work which is completely unnecessary for no additional
> benefit. Unless there's something I'm missing.
>

If nothing else, there is a maintainability benefit.

Instead of reusing code, you are adding code that duplicates an existing
process, which means there are more potential locations for error.

You also need to write separate tests that hit both common cases and edge
cases for the process.  If the process just returns the result of
matchingList(), then you basically are just testing the interface to the
function.  The internals of the function are tested with the tests for
matchingList().

Paul